settings.php

```diff
@@ -46,5 +46,12 @@
             get_string('settings:coursepageprintgrade_desc', 'mod_subcourse'),
             1
         ));
+
+        $settings->add(new admin_setting_configcheckbox(
+            'mod_subcourse/limittoenrolled',
+            get_string('settings:limittoenrolled', 'mod_subcourse'),
+            get_string('settings:limittoenrolled_desc', 'mod_subcourse'),
+            0
+        ));
```

Comment:
Related to the comment above, I am not fully sure this actually works as-is now. Were you able to measure that there is a performance issue with the existing call and was it fixed by this change?

Reply:
We have an instance here with > 400000 courses where using the limittoenrolled function (of the external function) is the only way to keep course searching functional.